src/runtime_src/core/common/api/xrt_module.cpp

```diff
@@ -1394,6 +1394,29 @@ class module_sram : public module_impl
   {
       return m_scratch_pad_mem;
   }
+
+  void
+  dump_scratchpad_mem()
+  {
+    if (m_scratch_pad_mem.size() == 0) {
+      xrt_core::message::send(xrt_core::message::severity_level::debug, "xrt_module",
+                              "preemption scratchpad memory is not available");
+      return;
+    }
+
```

Comment:
There is already a dump_bo function in this file, can we reuse that one?

Reply:
changed code to use existing function